examples/microprofile/grpc/src/main/resources/application.yaml

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2024 Oracle and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+server:
+  port: 0
+  tls:
+    trust:
+      keystore:
+        passphrase: "password"
+        trust-store: true
+        resource:
+          resource-path: "server.p12"
+    private-key:
+      keystore:
+        passphrase: "password"
+        resource:
+          resource-path: "server.p12"
+
+grpc:
+  channels:
+    string-channel:
+      port: 0
```

Comment:
How is the host & port discovered by the client? 
Also `grpc.channels` does not mention `client`, yet it seems to be a configuration for the client. Can you please explain this choice? I would expect `grpc.client.channels` or similar

Reply:
1. There's some ugly code that attempts to find the server port when running on the same VM and no port was specified in config. Only used for testing.
2. The use of `grpc.channels` instead of something like `grpc.client.channels` is carried over from 3.X. We can change that.

Reply:
Using `grpc.client.channels` in latest commit

Reply:
Removed ugly code to find server port in favor for new interface implemented by client proxies to dynamically set a new port. Tests updated.